LingoClash/LingoClash/ViewModels/Auth/LoginViewModel.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  LoginViewModel.swift
+//  LingoClash
+//
+//  Created by Kyle キラ on 14/3/22.
+//
+
+import Foundation
+import FirebaseAuth
+import FirebaseFirestore
+
+final class LoginViewModel {
+    
+    @Published var error: String?
+    
+    func login(email: String, password: String) {
+        
+        let fields = [
+            "email": email,
+            "password": password
+        ]
+        
+        // Validate fields
+        let error = validateFields(fields)
+        if let error = error {
+            self.error = error
+            return
+        }
+        
+        // Sign in
+        Auth.auth().signIn(withEmail: email, password: password) { [weak self] result, error in
```

Comment:
Perhaps if we have time, we can extract out the sign in logic etc into a wrapper around the authentication layer so that we can switch authentication methods easily (similar to how we are going to do it for the database)

Reply:
Yep I agree, we should have an authentication layer!